src/Http/HttpDownloadManager.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "Http/HttpDownloadManager.h"
+
+#include "OrbitBase/Promise.h"
+
+namespace orbit_http {
+using orbit_base::CanceledOr;
+using orbit_base::Future;
+using orbit_base::Promise;
+using orbit_base::StopToken;
+using orbit_http_internal::HttpDownloadOperation;
+
+orbit_base::Future<ErrorMessageOr<orbit_base::CanceledOr<void>>> HttpDownloadManager::Download(
+    std::string url, std::filesystem::path save_file_path, orbit_base::StopToken stop_token) {
+  orbit_base::Promise<ErrorMessageOr<orbit_base::CanceledOr<void>>> promise;
+  auto future = promise.GetFuture();
+
+  auto current_download_operation = new HttpDownloadOperation{
+      std::move(url), std::move(save_file_path), std::move(stop_token), &manager_};
+  download_operations_.push_back(current_download_operation);
```

Comment:
This has still the problem that we never remove elements from `download_operations_`. In other words, that's a memory leak.

There are 2 options:
1. Remove operations once they are done (from the finish handler)
2. Do the following in `~HttpDownloadManager()`:
```cpp
for (const auto& download_operation : findChildren<HttpDownloadOperation*>()) {
  download_operation->Abort();
}
```

The latter uses QObject's parent child relationship. That means `download_operations_` won't be needed - so  would prefer that.

Reply:
I like the second one!!! Thank you Henning!!!!!!!!!!!!!

Reply:
Let me know if it doesn't work.